### DISCUSSION TITLE
Filter data based on grafana's timerange

### DISCUSSION THREAD

**Discussion by adrianomazi** [grafana/grafana-csv-datasource#48] [Q&A] [answered]
Hi @marcusolsson , I'm testing the plugin and it's working great. But I'm wondering if I can filter the information based on the dashboard timerange (my file has only 1 date column using this format YYYY-MM-DD).

Let me know if you need more details.

**Comment by marcusolsson** [comment, marked as answer]
Great to hear! Unfortunately, it's currently not possible to filter based on time interval. The challenge is mostly on the UX side, when the CSV has multiple time fields (how do you configure which fields to filter by?). For CSV files with only one time field, we can probably filter it by default.

**Reply by adrianomazi** [reply]
Thank you for your reply. That in fact it's my doubt, how can I filter? Using regex or maybe using a variable? The date should follow some standard format?

My file has only 1 field, I have tried a few things but none of them worked, created regex, tried to build a expression, used the Name and Type as Date, etc.

**Reply by slhck** [reply]
I guess per #231 it is still not possible to filter by a time field?

For Elasticsearch datasources, one can select which field contains the timestamp, so I believe it should be possible from an UX perspective. In the case of a CSV it would likely be just a text field with the name of the column.

<img width="544" alt="image" src="https://github.com/grafana/grafana-csv-datasource/assets/582444/0c6bbd6c-31d3-43b1-89f8-ed1c7b70108a">
